src/borg/repository.py

```diff
@@ -1448,6 +1455,9 @@ def clean_old():
                         del self.fds[k]
 
         clean_old()
+        if self._write_fd is not None:
+            # without this, we have test failure now
+            self._write_fd.sync()
```

Comment:
this was needed now to avoid a test failure.
otherwise it could not read the MAGIC back from a recently created segment file.

Reply:
This is done a bit more than necessary, sufficient would be:
```
self._write_fd.f.flush()
```

But this is not available yet via SyncFile api.